Add cannode RTCMStream subscriber and fix RTCMStream Publisher

AlexKlimaj opened the pull request:
This PR adds an option to subscribe to RTCMStream messages on cannode GPS. Ardupilot publishes RTCMStream when using a fixed base but uses MoveBaselineData when using a moving base.

WARGRaph commented:
The CI complains because there is no RTCMStream.hpp in src/drivers/uavcannode/Subscribers. 
I took it from [here](https://github.com/PX4/PX4-Autopilot/blob/0a662ef22c706aaa3a188efd718a08c47de72c9a/src/drivers/uavcannode/Subscribers/RTCMStream.hpp) and compiled it again.
After loading it to the GPS unit I can confim that it reaches an rtk Fixed solution connected to an FC running ArduPilot and getting RTCM data from an NTRIP provider.

dagar commented:
I'm slightly worried about user facing configuration and how we'll know what's needed in different use cases.

Can we make a quick list of where we need RTCMStream vs MovineBaselineData?

For the CAN node I'm also wondering if they should be separately configurable. To avoid every user having to manually configure it might be less bad overall to have both RTCMStream and MovineBaselineData subscribers running by default. If both are present we could then automatically disable the other.

AlexKlimaj commented:
I just updated and pushed new params.
`CANNODE_SUB_RTCM` controls subscribing to RTCMStream or MovingBaselineData. There shouldn't be a case where a rover should be subscribed to both.
`CANNODE_PUB_MBD` controls publishing MovingBaselineData for moving bases.

In the case of a dual GPS setup with a moving base, the rover would set `CANNODE_SUB_RTCM` to 1 and `GPS_UBX_MODE` to 3.
The moving base would set `CANNODE_SUB_RTCM` to 0, `CANNODE_PUB_MBD` to 1, and `GPS_UBX_MODE` to 4.

I need to test a triple GPS setup with a Rover, Moving Base, and Fixed Base. In that case, the rover would set `CANNODE_SUB_RTCM` to 1 and `GPS_UBX_MODE` to 3.
The moving base would set `CANNODE_SUB_RTCM` to 2, `CANNODE_PUB_MBD` to 1, and `GPS_UBX_MODE` to 4. 
The flight controller would receive corrections from the fixed base and publish to RTCMStream by setting `UAVCAN_PUB_RTCM` to 1.

AlexKlimaj commented:
Triple F9P setup. QGC with Fixed Base -> Radio -> FC -> RTCMStream -> Moving Base -> MovingBaselineData -> Rover with Heading
https://review.px4.io/plot_app?log=649cf5a9-c2b0-4d76-8868-cca4bc9ab05c

mcsauder reviewed (approved):
This looks like a good improvement.

dagar commented on src/drivers/uavcannode/uavcannode_params.c:
```diff
@@ -61,10 +61,31 @@ PARAM_DEFINE_INT32(CANNODE_BITRATE, 1000000);
+ * Enable RTCM subscription
+ *
+ * @boolean
+ * @max 1
```
```suggestion
```

dagar commented on src/drivers/uavcannode/uavcannode_params.c:
```diff
@@ -61,10 +61,31 @@ PARAM_DEFINE_INT32(CANNODE_BITRATE, 1000000);
+ * Enable MovingBaselineData publication
+ *
+ * @boolean
+ * @max 1
```
```suggestion
```

dagar commented on src/drivers/uavcannode/Subscribers/RTCMStream.hpp:
```diff
@@ -0,0 +1,108 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```
```suggestion
 *   Copyright (c) 2022 PX4 Development Team. All rights reserved.
```
